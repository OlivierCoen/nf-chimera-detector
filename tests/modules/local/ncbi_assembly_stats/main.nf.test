nextflow_process {

    name "Test Process NCBI_ASSEMBLY_STATS"
    script "modules/local/ncbi_assembly_stats/main.nf"
    process "NCBI_ASSEMBLY_STATS"
    tag "ncbi_assembly_stats"

    test("Has genomes") {

        when {
            process {
                """
                input[0] = "307456"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Has NO genomes") {

        when {
            process {
                """
                input[0] = "Hydriviridae"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
            assert process.trace.failed().size() == 1
        }

    }

    test("Parent has genomes") {

        when {
            process {
                """
                input[0] = "Mamonoviridae"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
