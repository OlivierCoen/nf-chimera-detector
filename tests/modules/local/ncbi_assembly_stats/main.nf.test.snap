{
    "Has genomes": {
        "content": [
            {
                "0": [
                    [
                        "307456",
                        "133904"
                    ]
                ],
                "1": [
                    [
                        "GET_MEAN_ASSEMBLY_LENGTH",
                        "python",
                        "3.13.5"
                    ]
                ],
                "2": [
                    [
                        "GET_MEAN_ASSEMBLY_LENGTH",
                        "requests",
                        "2.32.4"
                    ]
                ],
                "families": [
                    [
                        "307456",
                        "133904"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T15:47:54.426828307"
    },
    "Parent has genomes": {
        "content": [
            {
                "0": [
                    [
                        "Mamonoviridae",
                        "244505"
                    ]
                ],
                "1": [
                    [
                        "GET_MEAN_ASSEMBLY_LENGTH",
                        "python",
                        "3.13.5"
                    ]
                ],
                "2": [
                    [
                        "GET_MEAN_ASSEMBLY_LENGTH",
                        "requests",
                        "2.32.4"
                    ]
                ],
                "families": [
                    [
                        "Mamonoviridae",
                        "244505"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T15:55:33.903618117"
    },
    "Has NO genomes": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "families": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T15:48:01.896530627"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        "307456",
                        "133904"
                    ]
                ],
                "1": [
                    [
                        "GET_MEAN_ASSEMBLY_LENGTH",
                        "python",
                        "3.13.5"
                    ]
                ],
                "2": [
                    [
                        "GET_MEAN_ASSEMBLY_LENGTH",
                        "requests",
                        "2.32.4"
                    ]
                ],
                "families": [
                    [
                        "307456",
                        "133904"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-24T15:47:05.783761999"
    }
}